microhapulator/profile.py

```diff
@@ -117,14 +117,14 @@ def rand_match_prob(self, freqs):
                 raise RandomMatchError(msg)
             result = freqs[(freqs.Marker == marker) & (freqs.Haplotype.isin(haplotypes))]
             if len(haplotypes) == 1:
-                p = 0.001
+                p = minfreq
                 if len(result) == 1:
                     pp = list(result.Frequency)[0]
                     if pp > 0.0:
                         p = pp
                 prob *= p * p
             else:
-                p, q = 0.001, 0.001
+                p, q = minfreq, minfreq
```

Comment:
Tangential change: replacing the hard-coded 0.001 value with a `minfreq` parameter for rare alleles. The `minfreq` parameter now has a default value of 0.01, which is more reasonable than the previous 0.001.